db/db_impl.cc

```diff
@@ -2474,22 +2474,22 @@ Status DBImpl::DeleteFilesInRanges(ColumnFamilyHandle* column_family,
             i, begin_key, end_key, &level_files, -1 /* hint_index */,
             nullptr /* file_index */);
         FileMetaData* level_file;
+        deleted_files.reserve(level_files.size());
+        edit.ReserveDelete(level_files.size());
         for (uint32_t j = 0; j < level_files.size(); j++) {
           level_file = level_files[j];
           if (level_file->being_compacted) {
             continue;
           }
-          if (deleted_files.find(level_file) != deleted_files.end()) {
-            continue;
-          }
-          if (!include_end && end != nullptr &&
+          // level files are sorted in order, so here only checks the end key of the last file
+          if (j == level_files.size() - 1 && !include_end && end != nullptr &&
```

Comment:
don't understand this change. we are iterating through the files and checking range bound one-by-one. why only the last file in the level needs to be checked?

Reply:
level files are all files overlaps with the [begin, end] in that level, and those files have no overlap between each other. Am I thinking wrong?